### PR TITLE
Dont polute the log on DAV emaillogin

### PR DESCRIPTION
**Pull request by rullzer**
To test:

1. setup a user `user1` and set the email to something
1. `curl -u <email>:<pass> <server>/remote.php/webdav -X PROPFIND`

Before: get a log entry saying login failed
After: No log entry

I also fixed that the thorttler was not increased when the email login was also invalid.

### CODE REVIEW

File: tests/lib/User/SessionTest.php [hunk 1]
```diff
@@ -29,6 +29,7 @@
 use OCP\Lockdown\ILockdownManager;
 use OCP\Security\ICrypto;
 use OCP\Security\ISecureRandom;
+use const ZipStreamer\INT64_LOW_MAP;
```
Comment (MorrisJobke) [inline tests/lib/User/SessionTest.php]: ❓

Comment (rullzer) [reply tests/lib/User/SessionTest.php]: yes good question :confused:

File: lib/private/User/Session.php [hunk 2]
```diff
@@ -400,17 +406,22 @@ public function logClientIn($user,
...
-			if($currentDelay === 0) {
-				$throttler->sleepDelay($request->getRemoteAddress(), 'login');
+				$this->logger->warning('Login failed: \'' . $user . '\' (Remote IP: \'' . \OC::$server->getRequest()->getRemoteAddress() . '\')', ['app' => 'core']);
+
+				$throttler->registerAttempt('login', $request->getRemoteAddress(), ['uid' => $user]);
+				if ($currentDelay === 0) {
+					$throttler->sleepDelay($request->getRemoteAddress(), 'login');
+				}
+				return false;
 			}
```
Comment (icewind1991) [inline lib/private/User/Session.php]: missing an `else {return true;}`?

**Review by icewind1991 — CHANGES_REQUESTED** [review]

File: lib/private/User/Session.php (hunk 2 above)
Comment (rullzer) [reply lib/private/User/Session.php]: See https://github.com/nextcloud/server/blob/ef127a30ec9acd07c32952bf8b97059b280f7168/lib/private/User/Session.php#L434

**Review by icewind1991 — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Pull request merged**